webapp/components/notifications/Notification/Notification.vue

```diff
@@ -54,23 +47,21 @@ export default {
     },
   },
   computed: {
-    resourceType() {
-      return this.post.id ? 'Post' : 'Comment'
+    from() {
+      return this.notification.from
     },
-    post() {
-      return this.notification.post || {}
-    },
-    comment() {
-      return this.notification.comment || {}
+    isComment() {
+      return this.from.__typename === 'Comment'
     },
     params() {
+      const post = this.isComment ? this.from.post : this.from
       return {
-        id: this.post.id || this.comment.post.id,
-        slug: this.post.slug || this.comment.post.slug,
+        id: post.id,
+        slug: post.slug,
       }
     },
     hashParam() {
-      return this.post.id ? {} : { hash: `#commentId-${this.comment.id}` }
+      return this.isComment ? { hash: `#commentId-${this.from.id}` } : {}
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-08-30T08:43:30Z" title="Friday, August 30th 2019, 10:43:30 am +02:00">Aug 30, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -54,23 +47,21 @@ export default {
     },
   },
   computed: {
-    resourceType() {
-      return this.post.id ? 'Post' : 'Comment'
+    from() {
+      return this.notification.from
     },
-    post() {
-      return this.notification.post || {}
-    },
-    comment() {
-      return this.notification.comment || {}
+    isComment() {
+      return this.from.__typename === 'Comment'
     },
     params() {
+      const post = this.isComment ? this.from.post : this.from
       return {
-        id: this.post.id || this.comment.post.id,
-        slug: this.post.slug || this.comment.post.slug,
+        id: post.id,
+        slug: post.slug,
       }
     },
     hashParam() {
-      return this.post.id ? {} : { hash: `#commentId-${this.comment.id}` }
+      return this.isComment ? { hash: `#commentId-${this.from.id}` } : {}
```


This whole thing gets an amazing easiness. 💃🏻

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-30T12:40:31Z" title="Friday, August 30th 2019, 2:40:31 pm +02:00">Aug 30, 2019</time>_

---


really great refactor!